src/infra/marketplace/marketplaceCmdRepo.go

```diff
@@ -0,0 +1,316 @@
+	dataFields []valueObject.MarketplaceInstallableItemDataField,
+) ([]valueObject.MarketplaceItemCmdStep, error) {
+	cmdStepsWithDataFields := []valueObject.MarketplaceItemCmdStep{}
+
+	dataFieldsMap := map[string]string{}
+	for _, dataField := range dataFields {
+		dataFieldKeyStr := dataField.Key.String()
+		dataFieldsMap[dataFieldKeyStr] = dataField.Value.String()
+	}
+
+	for _, cmdStep := range cmdSteps {
+		cmdStepStr := cmdStep.String()
+		cmdStepDataFieldKeys, _ := infraHelper.GetAllRegexGroupMatches(
+			cmdStepStr,
+			`%(.*?)%`,
+		)
+
+		for _, cmdStepDataFieldKey := range cmdStepDataFieldKeys {
+			dataFieldValue := dataFieldsMap[cmdStepDataFieldKey]
+			cmdStepWithDataFieldStr := strings.ReplaceAll(
+				cmdStepStr,
+				"%"+cmdStepDataFieldKey+"%",
+				dataFieldValue,
+			)
+			cmdStepStr = cmdStepWithDataFieldStr
+		}
+
+		cmdStepWithDataField, _ := valueObject.NewMarketplaceItemCmdStep(cmdStepStr)
+		cmdStepsWithDataFields = append(cmdStepsWithDataFields, cmdStepWithDataField)
+	}
+
+	return cmdStepsWithDataFields, nil
+}
+
+func (repo *MarketplaceCmdRepo) runCmdSteps(
+	catalogCmdSteps []valueObject.MarketplaceItemCmdStep,
+	catalogDataFields []valueObject.MarketplaceCatalogItemDataField,
+	receivedDataFields []valueObject.MarketplaceInstallableItemDataField,
+	installDir valueObject.UnixFilePath,
+	installUuid string,
+) error {
+	installDirDataFieldKey, _ := valueObject.NewDataFieldKey("installDirectory")
+	installDirDataFieldValue, _ := valueObject.NewDataFieldValue(installDir.String())
+	installDirDataField, _ := valueObject.NewMarketplaceInstallableItemDataField(
+		installDirDataFieldKey,
+		installDirDataFieldValue,
+	)
+
+	installUuidDataFieldKey, _ := valueObject.NewDataFieldKey("installUuid")
+	installUuidDataFieldValue, _ := valueObject.NewDataFieldValue(installUuid)
+	installUuidDataField, _ := valueObject.NewMarketplaceInstallableItemDataField(
+		installUuidDataFieldKey,
+		installUuidDataFieldValue,
+	)
+
+	receivedDataFields = append(
+		receivedDataFields,
+		installDirDataField,
+		installUuidDataField,
+	)
+
+	missingCatalogOptionalDataFields, err := repo.getMissingOptionalDataFields(
+		receivedDataFields,
+		catalogDataFields,
+	)
+	if err != nil {
+		return err
+	}
+	receivedDataFields = slices.Concat(
+		receivedDataFields,
+		missingCatalogOptionalDataFields,
+	)
+
+	log.Printf("ReceivedDataFields: %+v", receivedDataFields)
+
+	preparedCmdSteps, err := repo.replaceCmdStepsPlaceholders(
+		catalogCmdSteps,
+		receivedDataFields,
+	)
+	if err != nil {
+		return errors.New("ParseCmdStepWithDataFieldsError: " + err.Error())
+	}
+
+	for preparedCmdStepIndex, preparedCmdStep := range preparedCmdSteps {
+		preparedCmdStepStr := preparedCmdStep.String()
+		_, err = infraHelper.RunCmdWithSubShell(preparedCmdStepStr)
+		if err != nil {
+			preparedCmdStepIndexStr := strconv.Itoa(preparedCmdStepIndex)
+			return errors.New(
+				"RunCmdStepError (" + preparedCmdStepIndexStr + "): " + err.Error(),
+			)
+		}
```

Comment:
Cara, você ainda tem muita dificuldade de entender contexto. Veja, você está dentro de um contexto de for loop trabalhando com a variável "preparedCmdSteps". Não há necessidade de você chamar a iteration de "preparedCmdStepIndex" e "preparedCmdStep", basta chamar de "stepIndex" e "cmdStep". Essa verbosidade excessiva tua tem que acabar cara, olha o tamanho da variável na linha 200. Zero necessidade desse nível de verbosidade dentro de um for específico de um método ainda mais específico.

Reply:
Ajustado.
Vou parar de dar essas bolas foras!